boards/arm/arduino_opta_m4/doc/index.rst

```diff
@@ -0,0 +1,170 @@
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| IPM       | on-chip    | virtual mailbox based on HSEM       |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+The default configuration per core can be found in the defconfig files:
+``boards/arm/arduino_opta_m4/arduino_opta_m4_defconfig``
+
+Pin Mapping
+===========
+
+ARDUINO OPTA M4 has access to the 9 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For mode details please refer to `ARDUINO-OPTA website`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- Status LD1 : PI0
+- Status LD2 : PI1
+- Status LD3 : PI3
+- Status LD4 : PH15
+- USER : PE4
+
+System Clock
+============
+
+The STM32H747I System Clock can be driven by an internal or external oscillator,
+as well as by the main PLL clock. By default, the CPU2 (Cortex-M4) System clock
+is driven at 200MHz. PLL clock is feed by a 25MHz high speed external clock.
+
+Resources sharing
+=================
+
+The dual core nature of STM32H747 SoC requires sharing HW resources between the
+two cores. This is done in 3 ways:
+
+- **Compilation**: Clock configuration is only accessible to M7 core. M4 core only
+  has access to bus clock activation and deactivation.
+- **Static pre-compilation assignment**: Peripherals such as a UART are assigned in
+  devicetree before compilation. The user must ensure peripherals are not assigned
+  to both cores at the same time.
+- **Run time protection**: Interrupt-controller and GPIO configurations could be
+  accessed by both cores at run time. Accesses are protected by a hardware semaphore
+  to avoid potential concurrent access issues.
+
+Programming and Debugging
+*************************
+
+Applications for the ``arduino_opta_m4`` uses the regular Zephyr build commands
+See :ref:`build_an_application` for more information about application builds.
+
+Flashing
+========
+
+Flashing operation will depend on the target to be flashed and the SoC
+option bytes configuration. The OPTA has an DFU capable bootloader which
+can be accessed by connecting the device to the USB, and then pressing
+short time the RESET two times, the RESET-LED on the board will fade
+indicating the board is in bootloader.
+
+By default:
+
+  - CPU2 (Cortex-M4) boot address is set to 0x81000000 (OB: BOOT_CM4_ADD0)
```

Comment:
The PLC binary running on the M7 could grow a bit in the future, maybe it's better to dedicate a bit less flash to the M4? I think 512K would be sensible. Then on the Arduino side we can just call bootM4() https://github.com/arduino/ArduinoCore-mbed/blob/main/variants/OPTA/variant.cpp#L380-L384  with the right address

Reply:
Yes, I noticed in Arduino IDE it is possible to do flash partitioning in order to leave 512k of flash memory to the M4 instead of equal split.

Reply:
Moved the flash1 reagion 512K ahead, and left 1.5MB for Arduino and 512K upper for the M4 + Zephyr.

Reply:
Super, I've seen the change in the dts but the documentation still uses 0x81000000 as default address